### PR TITLE
Add job/pull-ci-openshift-release-sync-rover-groups

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @openshift/test-platform 

/hold

wait for rehearsal

### CODE REVIEW

File: ci-operator/jobs/openshift/release/openshift-release-master-presubmits.yaml [hunk 1]
```diff
@@ -1371,6 +1371,38 @@ presubmits:
...
+    branches:
+    - ^master$
+    - ^master-
+    cluster: build03
+    context: ci/prow/sync-rover-groups
+    decorate: true
+    labels:
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-release-sync-rover-groups
+    optional: true
```
Comment (petr-muller) [inline ci-operator/jobs/openshift/release/openshift-release-master-presubmits.yaml:1385]: Let's make it:

```
always_run: false
run_if_changed: core-services/sync-rover-groups/_config.yaml
```

Context is in https://github.com/openshift/release/pull/25651:

> alvaroaleman pointed out that o/release presubmits likely generate a lot of GH API traffic in crier. We could optimize this with a reasonable usage of run_if_changed: the repository is partitioned quite nicely and most presubmits deal with only one partition. This also lessens the load in CI overall, makes o/release PRs less cluttered...

Comment (hongkailiu) [reply ci-operator/jobs/openshift/release/openshift-release-master-presubmits.yaml:1385]: We finally reach the point where we do not afford running everything on every PR.

But this presubmit checks the violations such as define groups in the `clusters` folder.
Config file should not matter when `--validate-subjects`.

Let me remove it.

**Review by smg247 — APPROVED** [review]

**Pull request merged**